relays/substrate-client/src/lib.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright 2019-2020 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges Common.
+
+// Parity Bridges Common is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Bridges Common is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges Common.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Tools to interact with (Open) Ethereum node using RPC methods.
+
+#![warn(missing_docs)]
+
+mod chain;
+mod client;
+mod error;
+mod rpc;
+
+#[cfg(feature = "millau")]
+pub mod millau;
+
+#[cfg(feature = "rialto")]
+pub mod rialto;
+
+pub use crate::chain::{Chain, TransactionSignScheme};
+pub use crate::client::Client;
+pub use crate::error::{Error, Result};
+
+#[cfg(feature = "millau")]
+pub use crate::millau::Millau;
+
+#[cfg(feature = "rialto")]
+pub use crate::rialto::Rialto;
```

Comment:
Isn't it possible to split this into it's own crates? Instead of activating a feature, the crate would simply have two deps: `substrate-client` and the runtime it uses. If the crate is generic it could just import `substrate-client` and then accept any `Chain` object.
Mid-term I think it would make sure we don't hardcode even more runtime-specific things into the relay code.

Reply:
It is possible - I just wanted to avoid too many crates. So idea is that we have basic crate with basic definitions + for basic chain that we use. Everyone else may implement their own `trait Chain` to get `Client` functionality. Happy to split it if you think it's better.

> Mid-term I think it would make sure we don't hardcode even more runtime-specific things into the relay code.

I don't get it - how split will help? :)

Reply:
> To make sure everything actually works properly you should run multiple separate builds with different combination of feature flags

Ok, this makes sense - I'll extract to separate crates then

Reply:
done